provider-example/src/aead.rs

```diff
@@ -0,0 +1,158 @@
+use chacha20poly1305::{AeadInPlace, KeyInit, KeySizeUser};
+use rustls::crypto::cipher;
+use rustls::internal::msgs::base::Payload; // FIXME
```

Comment:
What's the `// FIXME` in relation to? Should it be addressed pre-merge?

Reply:
ah, i forgot about that. have done the requisite refactoring to remove the need for that type to be used outside the crate.